ale_python_interface/ale_c_wrapper.cpp

```diff
@@ -1,3 +1,20 @@
-//all code is currently in the .h file
 #include "ale_c_wrapper.h"
 
+#include <cstring>
+#include <string>
+
+void encodeState(ALEState *state, char *buf) {
+	std::string str = state->serialize();
+
+	memcpy(buf, str.data(), str.length());
+}
+
+int encodeStateLen(ALEState *state) {
```

Comment:
That's true, but this way allows wrappers to avoid dynamic allocations if they want.